test/controllers/users.test.js

```diff
@@ -0,0 +1,99 @@
+const request = require('supertest');
+
+const app = require('../../app.js');
+const userFactory = require('../factory/users');
+const {
+  CONFLICT_CODE,
+  CREATED_CODE,
+  DEFAULT_HEADER,
+  UNPROCESSABLE_ENTITY_CODE,
+  USERS_PATH
+} = require('../constants/');
+const {
+  conflictErrorResponse,
+  correctWoloxEmail,
+  randomMissingParams,
+  user: userTestInfo,
+  userCreatedResponse,
+  wrongPasswords,
+  wrongWoloxEmail
+} = require('../fixtures/users');
+
+describe(`SignUp endpoint POST ${USERS_PATH}`, () => {
+  describe('satisfactory cases - suite', () => {
+    test('success json reponse (when users was created)', () =>
+      request(app)
+        .post(USERS_PATH)
+        .set(DEFAULT_HEADER)
+        .send(userTestInfo)
+        .then(res => expect(res.body).toStrictEqual(userCreatedResponse)));
+    test('user successfully created', () =>
+      request(app)
+        .post(USERS_PATH)
+        .set(DEFAULT_HEADER)
+        .send(userTestInfo)
+        .then(res => expect(res.statusCode).toBe(CREATED_CODE)));
+    test.each(correctWoloxEmail)('user created with a valid wolox email: %p', randomTestData =>
+      request(app)
+        .post(USERS_PATH)
+        .set(DEFAULT_HEADER)
+        .send(randomTestData)
+        .then(res => expect(res.statusCode).toBe(CREATED_CODE))
+    );
+  });
+  describe('failed cases - suite', () => {
+    test('failed with an email already exists', () =>
+      userFactory
+        .create()
+        .then(() =>
+          request(app)
+            .post(USERS_PATH)
+            .set(DEFAULT_HEADER)
+            .send(userTestInfo)
+        )
+        .then(res => expect(res.statusCode).toBe(CONFLICT_CODE)));
```

Comment:
why aren't you using `.catch` here instead of `.then`?

Reply:
this comment applies to all failed cases tests

Reply:
I generally don't put catch in tests files. Thanks by feedback!

Reply:
Done!